test/nonregression/pipelines/test_run_pipelines_dwi.py

```diff
@@ -52,6 +52,83 @@ def test_dwi_preprocessing_using_t1(cmdopt, tmp_path):
     run_dwi_preprocessing_using_t1(input_dir, tmp_dir, ref_dir, working_dir)
 
 
+@pytest.mark.slow
+def test_dwi_compute_reference_b0(cmdopt, tmp_path):
+    """Test step 1 of pipeline DWIPreprocessingUsingPhaseDiff.
+
+    This is a slow test. Expect approximately a 2.5 hours runtime.
+
+    We use the same input files as for the DWIPreprocessingUsingPhaseDiff pipeline.
+
+    However, since we are testing the step 1 in isolation, we don't have the
+    BIDS querying logic available to us. The compute_reference_b0 pipeline is
+    not BIDS-aware, it just expects three files:
+        - the DWI image
+        - the associated b-values
+        - and the associated b-vectors
+
+    These files are directly available in the input_dir (no BIDS structure, files
+    are directly at the root level).
+
+    It also needs the phase encoding direction and total readout time which are
+    available in the JSON file (also available in the input_dir).
+    In the main pipeline, the input node is performing the extraction of these
+    metadata from the JSON file. Here, we perform this extraction directly with
+    the `extract_metadata_from_json` and `bids_dir_to_fsl_dir`functions.
+
+    The compute_reference_b0 workflow produces two outputs that are compared
+    against their reference values:
+        - The reference B0 volume
+        - The brain mask computed on the B0 volume
+    """
```

Comment:
Same comment as before. Could you get this docstring to be numpydocs compliant ?

Reply:
Hmmm, since this is a test there isn't really a standard docstring format.
I have written a lot of things because these tests are a bit different from the usual non-regression tests in clinica which we traditionally write only for full user-facing pipelines.

The sub-workflows I wish to test through these tests don't have the BIDS reading and CAPS writing capabilities of clinica's user facing pipelines. They simply expect a bunch of file paths and parameters as inputs, they do some computations writing files in temporary locations, and make some of them available as outputs.
By adding a writing node with conditional connections, the pipeline can potentially write to the test's `tmp_path`, and the output can then be compared to the reference.

This doesn't seem trivial to me, hence the verbose docstring. Let me know if it doesn't feel clear to you or if you have ideas to clarify things.

Reply:
okay, that sounds good, I wasn't sure about the docstring format since it is, as you mentionned,  a test !